internal/rbd/rbd_util.go

```diff
@@ -426,9 +426,15 @@ func flattenClonedRbdImages(ctx context.Context, snaps []snapshotInfo, pool, mon
 		util.ErrorLog(ctx, "failed to open connection %s; err %v", rv, err)
 		return err
 	}
-	for _, s := range snaps {
-		if s.Namespace.Type == "trash" {
-			rv.RbdImageName = s.Namespace.OriginalName
+
+	for _, snapInfo := range snaps {
+		// get original snap name for the snapshots in trash namespace
+		origSnapName, err := rv.getOrigSnapName(snapInfo.Id)
+		if err != nil {
+			return err
+		}
+		if origSnapName != "" {
+			rv.RbdImageName = origSnapName
 			err = rv.flattenRbdImage(ctx, cr, true, rbdHardMaxCloneDepth, rbdSoftMaxCloneDepth)
```

Comment:
we need to flatten the images only if its in the trash.

Reply:
Yes, that's what we are doing because origSnapName will be no-empty only for trash snapshots